crates/client/l1-messages/src/contract.rs

```diff
@@ -0,0 +1,49 @@
+use ethers::contract::abigen;
+use ethers::types::U256;
+use mp_felt::{Felt252Wrapper, Felt252WrapperError};
+use mp_transactions::HandleL1MessageTransaction;
+use starknet_api::transaction::Fee;
+
+use crate::error::L1MessagesWorkerError;
+
+abigen!(
+    L1Contract,
+    r"[
+	event LogMessageToL2(address indexed fromAddress, uint256 indexed toAddress, uint256 indexed selector, uint256[] payload, uint256 nonce, uint256 fee)
+]"
+);
+
+impl LogMessageToL2Filter {
+    pub fn try_get_fee(&self) -> Result<Fee, L1MessagesWorkerError> {
+        // Check against panic
+        // https://docs.rs/ethers/latest/ethers/types/struct.U256.html#method.as_u128
+        if self.fee > U256::from_big_endian(&(u128::MAX.to_be_bytes())) {
+            Err(L1MessagesWorkerError::ToFeeError)
+        } else {
+            Ok(Fee(self.fee.as_u128()))
+        }
+    }
+}
+
+impl TryFrom<&LogMessageToL2Filter> for HandleL1MessageTransaction {
+    type Error = L1MessagesWorkerError;
```

Comment:
I would create an error enum dedicated to this conversion, with different variants:
- InvalidContractAddress
- InvalidEntryPointSelector
- InvalidNonce
- InvalidCalldata

Each one containing the type of error that could have cause it

Reply:
Fair point!